src/inputs/DateFields/DateFieldBase.tsx

```diff
@@ -276,11 +306,15 @@ export function DateFieldBase(props: DateRangeFieldBaseProps | DateSingleFieldBa
           if (v) {
             setInputValue(v);
             // If changing the value directly (vs using the DatePicker), then we always use the short format
-            const parsed = mode === "range" ? parseDateRange(v, dateFormats.short) : parseDate(v, dateFormats.short);
+            const parsed = isRangeMode ? parseDateRange(v, dateFormats.short) : parseDate(v, dateFormats.short);
             onChange(parsed);
           }
+          // User has deleted all text in field
+          else if (v === undefined) {
+            setInputValue("")
+          }
```

Comment:
EdgeCase bug fix
Currently DateFieldBase actually didn't allow users to clear out text completely so technically if a user were to have a date from months `jan-sep` they couldnt manually change it to anything `oct-sept` since the first `0` char couldn't be removed.